lib/spack/spack/spec_set.py

```diff
@@ -101,7 +101,8 @@ def _compiler_specs(self, data):
         the current platform are returned.
         """
         # get usable compilers for current platform.
-        arch = ArchSpec(str(sarch.platform()), 'default_os', 'default_target')
+        platform = (str(sarch.platform()), 'default_os', 'default_target')
```

Comment:
There should really be a shortcut for this, e.g., `ArchSpec.default()` or something.